File: typescript/tracing/langchain/00_quickstart.ts
import { HumanMessage, SystemMessage } from "@langchain/core/messages";
import { FakeListChatModel } from "@langchain/core/utils/testing";
import { LangChainInstrumentor } from "@respan/instrumentation-langchain";
import { Respan } from "@respan/respan";
import dotenv from "dotenv";
import { fileURLToPath } from "node:url";

dotenv.config({
  path: fileURLToPath(new URL("../../../.env", import.meta.url)),
  quiet: true,
});

export async function langchainInstrumentationQuickstart(): Promise<void> {
  const apiKey = process.env.RESPAN_API_KEY;
  const langchain = new LangChainInstrumentor();
  let respan: Respan | undefined;

  if (apiKey) {
    respan = new Respan({
      apiKey,
      baseURL: process.env.RESPAN_BASE_URL,
      appName: "typescript-langchain-quickstart",
      instrumentations: [langchain],
      logLevel: "error",
      silenceInitializationMessage: true,
    });
    await respan.initialize();
  } else {
    console.log("RESPAN_API_KEY is not set; running locally without exporting spans.");
  }

  const model = new FakeListChatModel({
    responses: ["Hello from a traced TypeScript LangChain run."],
  });
  const config = apiKey
    ? langchain.addCallback({
        runName: "quickstart",
        tags: ["respan-langchain-example", "quickstart"],
        metadata: {
          example: "quickstart",
          custom_identifier: process.env.RESPAN_EXAMPLE_RUN_ID,
        },
      })
    : {
        runName: "quickstart",
        tags: ["respan-langchain-example", "quickstart"],
        metadata: {
          example: "quickstart",
          custom_identifier: process.env.RESPAN_EXAMPLE_RUN_ID,
        },
      };

  try {
    const response = await model.invoke(
      [
        new SystemMessage("Reply in one short sentence."),
        new HumanMessage("Say hello to Respan tracing."),
      ],
      config,
    );
    console.log(response.content);
  } finally {
    await respan?.shutdown().catch(() => undefined);
  }
}

if (import.meta.url === `file://${process.argv[1]}`) {
  await langchainInstrumentationQuickstart();
}

File: python/tracing/langchain/00_quickstart.py
"""Hello-world quickstart for Respan LangChain instrumentation.

Run:
    RESPAN_API_KEY=your_respan_api_key python 00_quickstart.py

The example uses a fake LangChain chat model, so it does not need an OpenAI key.
When RESPAN_API_KEY is set, the LangChain run is exported to Respan.
"""

from __future__ import annotations

from _shared import init_telemetry, tracing_config
from langchain_core.language_models.fake_chat_models import FakeListChatModel
from langchain_core.messages import HumanMessage, SystemMessage


def langchain_instrumentation_quickstart() -> None:
    init_telemetry("langchain-quickstart")

    model = FakeListChatModel(responses=["Hello from a traced LangChain run."])
    response = model.invoke(
        [
            SystemMessage(content="Reply in one short sentence."),
            HumanMessage(content="Say hello to Respan tracing."),
        ],
        config=tracing_config("hello_world"),
    )
    print(response.content)


if __name__ == "__main__":
    langchain_instrumentation_quickstart()
